tutorials/analyze-cryptocurrency-data.md

```diff
@@ -0,0 +1,624 @@
+
+In order to get data from CryptoCompare, you’ll need to [obtain an API key][cryptocompare-apikey]. 
+For this analysis, the free key should be plenty.
+
+The script consists of five parts:
+
+- Importing the necessary Python libraries in order to complete the data extraction
+- Populate the `currency_info` table with a list of coin names
+- Get the historical Bitcoin (BTC) prices in 4198 other cryptocurrencies and populate the `crypto_prices` table
+- Get historical Bitcoin prices in different fiat currencies to populate `btc_prices`
+- Get historical Ethereum prices in different fiat currencies to populate `eth_prices`
+
+Here's the full Python script, which you can also [:DOWNLOAD_LINK: download][dataset-creation]:
+
+```python
+#####################################################################
+#1. Import library and setup API key
+#####################################################################
+import requests
+import json
+import csv
+from datetime import datetime
+
+apikey = 'YOUR_CRYPTO_COMPARE_API_KEY'
+#attach to end of URLstring
+url_api_part = '&api_key=' + apikey
+
+#####################################################################
+#2. Populate list of all coin names
+#####################################################################
+#URL to get a list of coins from cryptocompare API
+URLcoinslist = 'https://min-api.cryptocompare.com/data/all/coinlist'
+
+#Get list of cryptos with their symbols
+res1 = requests.get(URLcoinslist)
+res1_json = res1.json()
+data1 = res1_json['Data']
+symbol_array = []
+cryptoDict = dict(data1)
+
+#write to CSV
+with open('coin_names.csv', mode = 'w') as test_file:
+   test_file_writer = csv.writer(test_file, delimiter = ',', quotechar = '"', quoting=csv.QUOTE_MINIMAL)
+   for coin in cryptoDict.values():
+       name = coin['Name']
+       symbol = coin['Symbol']
+       symbol_array.append(symbol)
+       coin_name = coin['CoinName']
+       full_name = coin['FullName']
+       entry = [symbol, coin_name]
+       test_file_writer.writerow(entry)
+print('Done getting crypto names and symbols. See coin_names.csv for result')
+
+#####################################################################
+#3. Populate historical price for each crypto in BTC
+#####################################################################
+#Note: this part might take a while to run since we're populating data for 4k+ coins
+#counter variable for progress made
+progress = 0
+num_cryptos = str(len(symbol_array))
+for symbol in symbol_array:
+   # get data for that currency
+   URL = 'https://min-api.cryptocompare.com/data/histoday?fsym='+ symbol +'&tsym=BTC&allData=true' + url_api_part
+   res = requests.get(URL)
+   res_json = res.json()
+   data = res_json['Data']
+   # write required fields into csv
+   with open('crypto_prices.csv', mode = 'a') as test_file:
+       test_file_writer = csv.writer(test_file, delimiter = ',', quotechar = '"', quoting=csv.QUOTE_MINIMAL)
```

Comment:
Slightly long (see comment above)

Reply:
Other lines starting with `URL = ` and `test_file_writer = ` below are the same.